rsciio/jobin_yvon/api.py

```diff
@@ -0,0 +1,819 @@
+            "ND Filter",
+            "Laser (nm)",
+            "Spectro (nm)",
+            "Hole",
+            "Laser Pol. (°)",
+            "Raman Pol. (°)",
+            "X (µm)",
+            "Y (µm)",
+            "Z (µm)",
+            "Full time(s)",
+            "rotation angle (rad)",
+            "Windows",
+        ]
+
+        change_to_second_value = [
+            "Objective",
+            "Grating",
+            "ND Filter",
+            "Laser (nm)",
+            "Spectro (nm)",
+        ]
+
+        ## use second extracted value
+        for key in change_to_second_value:
+            try:
+                self.original_metadata["experimental_setup"][
+                    key
+                ] = self.original_metadata["experimental_setup"][key]["2"]
+            except KeyError:
+                pass
+
+        ## use first extracted value
+        for key, value in self.original_metadata["experimental_setup"].items():
+            if isinstance(value, dict):
+                # only if there is an entry/value
+                if bool(value):
+                    self.original_metadata["experimental_setup"][
+                        key
+                    ] = self.original_metadata["experimental_setup"][key]["1"]
+
+        for key, value in self.original_metadata["date"].items():
+            if isinstance(value, dict):
+                if bool(value):
+                    self.original_metadata["date"][key] = self.original_metadata[
+                        "date"
+                    ][key]["1"]
+
+        for key, value in self.original_metadata["file_information"].items():
+            if isinstance(value, dict):
+                if bool(value):
+                    self.original_metadata["file_information"][
+                        key
+                    ] = self.original_metadata["file_information"][key]["1"]
+
+        ## convert strings to float
+        for key in convert_to_numeric:
+            try:
+                self.original_metadata["experimental_setup"][key] = float(
+                    self.original_metadata["experimental_setup"][key]
+                )
+            except KeyError:
+                pass
+
+        ## move the unit from grating to the key name
+        new_grating_key_name = "Grating (gr/mm)"
+        try:
+            self.original_metadata["experimental_setup"][
+                new_grating_key_name
+            ] = self.original_metadata["experimental_setup"]["Grating"]
+            del self.original_metadata["experimental_setup"]["Grating"]
+        except KeyError:
+            pass  # pragma: no cover
+
+        ## add percentage for filter key name
+        new_filter_key_name = "ND Filter (%)"
+        try:
+            self.original_metadata["experimental_setup"][
+                new_filter_key_name
+            ] = self.original_metadata["experimental_setup"]["ND Filter"]
+            del self.original_metadata["experimental_setup"]["ND Filter"]
+        except KeyError:
```

Comment:
To me, this seems like an overly complicated way of writing

```python
self.original_metadata["experimental_setup"]["ND Filter (%)"] = \ 
    self.original_metadata["experimental_setup"].pop("ND Filter")
```

If you consider adding a default to the field such as NA, the try / except block could be removed altogether.

Reply:
I simplified this code block as you suggested. But I don't want to use default values for the dict. As this would complicate the metadata mapping.

Reply:
Ok :)